www/guides/react.md

```diff
@@ -0,0 +1,373 @@
+
+```diff
+import React, { useState, useEffect } from 'react';
++ import logo from './logo.svg';
+
+function App() {
+  // Create the count state.
+  const [count, setCount] = useState(0);
+  // Create the counter (+1 every second).
+  useEffect(() => {
+    const timer = setTimeout(() => setCount(count + 1), 1000);
+    return () => clearTimeout(timer);
+  }, [count, setCount]);
+  // Return the App component.
+  return (
+    <div className="App">
+      <header className="App-header">
++        <img src={logo} className="App-logo" alt="logo" />
+```
+
+Image: the React logo (a blue atom) is now at the top of the page
+
+The project already has index.css for global styles, but component CSS can help keep code organized. Snowpack knows to only load this CSS when displaying the `App` component, which means it's not loading unnecessary CSS.
+
+Create `src/App.css` and add this CSS:
+
+```css
+.App {
+  text-align: center;
+}
+.App code {
+  background: #fff3;
+  padding: 4px 8px;
+  border-radius: 4px;
+}
+.App p {
+  margin: 0.4rem;
+}
+.App-logo {
+  height: 40vmin;
+  pointer-events: none;
+}
+
+@media (prefers-reduced-motion: no-preference) {
+  .App-logo {
+    animation: App-logo-spin infinite 20s linear;
+  }
+}
+
+.App-header {
+  background-color: #282c34;
+  min-height: 100vh;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  font-size: calc(10px + 2vmin);
+  color: white;
+}
+
+.App-link {
+  color: #61dafb;
+}
+
+@keyframes App-logo-spin {
+  from {
+    transform: rotate(0deg);
+  }
+  to {
+    transform: rotate(360deg);
+  }
+}
+```
+
+To use it, import it `App.jsx` with
+
+```diff
+import logo from './logo.svg';
++import './App.css';
+```
+
+Image: screenshot or GIF. The page now has centered items, a grey background, styled fonts, and the React logo has an animation that rotates it
+
+You may notice the page has a white border, which doesn't look so great. That's due to a global style. Go to `index.css` and edit the CSS to remove it:
+
+```diff
+/* Add CSS styles here! */
+body {
+  font-family: sans-serif;
+-  margin: 2em 1em;
++  margin: 0;
+  line-height: 1.5em;
+}
+```
+
+Image:
+
+## Adding React Fast Refresh with a Snowpack plugin
+
+[React Fast Refresh](https://reactnative.dev/docs/fast-refresh)? What's that? It's a developer tool that makes it much easier to see how updates to your code work in the browser. React projects are often interactive and include state. For example, this project you're building has a state that is the amount of time on the page. When developing with state it's useful not to lose it while you edit code. React Fast Refresh shows you updates without refreshing the entire page. Showing you how to add this is also a good intro to Snowpack plugins. Snowpack starts with a minimal setup with the perspective that you can add what you need through the plugin system.
```

Comment:
Running out of time, but would love to take a pass on this section later. Overall I like this new approach though.

Reply:
Love the conversational tone here